model/lifecycle_test.go

```diff
@@ -1452,6 +1452,56 @@ func TestCreateTaskGroup(t *testing.T) {
 	assert.Contains(tasks[2].DependsOn[0].TaskId, "example_task_2")
 }
 
+func TestGetTaskIdTable(t *testing.T) {
+	require.NoError(t, db.Clear(task.Collection))
+
+	v := &Version{
+		Id:         "v0",
+		Requester:  evergreen.RepotrackerVersionRequester,
+		Revision:   "abcde",
+		CreateTime: time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC),
+	}
+
+	p := &Project{
```

Comment:
I was intending patch_optional tasks would be captured by the second test: a patch_optional task will be defined in the project but won't be part of newPairs since it's excluded by the dependencyIncluder.

I added an explanatory comment to the test.